compat.py

```diff
@@ -62,3 +70,60 @@ def escape_string(s):
                 result += chr(c)
         return result
 
+
+def run_in_subprocess(builder_function):
```

Comment:
I think the `compat` module was meant only for Python 2/3 compatibility definitions. This should maybe go in `methods`?

Reply:
This is more of platform compatibility. Maybe a new module would work best. Something like `plat_compat.py` or so.

Reply:
I don't know, `methods.py` already has some platform-specific methods. Maybe you could make a new `platform_methods.py` module and move the relevant `methods.py` methods there too.

Reply:
Good idea, but that would be unrelated to the scope of this PR and the ticket behind. 

To make it clear I suggest adding `platform_methods.py` with the new functions introduced here. Then we can move the rest of functions between the modules in a separate PR.

Reply:
Fine by me :)